### ISSUE TITLE
No version of staticcheck in gha and differences between local and gha check.

### ISSUE THREAD

**Issue by tsknadaj** [dominikh/go-tools#1160]
Hello,

we have our own github-actions for go repos and i added staticcheck into it but very strange things is between check on gha and local check.
On github-actions to install staticcheck i use command `GO111MODULE=off go get honnef.co/go/tools/cmd/staticcheck`

**GHA:**
- The output of 'staticcheck -version' : `(no version)`
- The output of 'staticcheck -debug.version' :
```
(no version)

Compiled with Go version: go1.17
Built without Go modules
```
- The output of 'go version' : `go version go1.17 linux/amd64`
- Exactly which command you ran: `staticcheck -checks="all","-ST1000","-ST1003","-SA1019" ./...`
- Output of the command and what's wrong with the output: on tests file few `should use suite.T().Logf(...) instead of suite.T().Log(fmt.Sprintf(...)) (S1038)`

**local:**
- The output of 'staticcheck -version' : `staticcheck 2021.1.2 (v0.2.2)`
- The output of 'staticcheck -debug.version' :
```
staticcheck 2021.1.2 (v0.2.2)

Compiled with Go version: go1.17.1
Main module:
        honnef.co/go/tools@v0.2.2 (sum: h1:MNh1AVMyVX23VUHE2O27jm6lNj3vjO5DexS4A1xvnzk=)
Dependencies:
        github.com/BurntSushi/toml@v0.3.1 (sum: h1:WXkYYl6Yr3qBf1K79EBnL4mak0OimBfB0XUf9Vl28OQ=)
        golang.org/x/mod@v0.4.2 (sum: h1:Gz96sIWK3OalVv/I/qNygP42zyoKp3xptRVCWRFEBvo=)
        golang.org/x/sys@v0.0.0-20210615035016-665e8c7367d1 (sum: h1:SrN+KX8Art/Sf4HNj6Zcz06G7VEz+7w9tdXTPOZ7+l4=)
        golang.org/x/tools@v0.1.2 (sum: h1:kRBLX7v7Af8W7Gdbbc908OJcdgtK8bOz9Uaj8/F1ACA=)
        golang.org/x/xerrors@v0.0.0-20200804184101-5ec99f83aff1 (sum: h1:go1bK/D/BFZV2I8cIQd1NKEZ+0owSTG1fDTci4IqFcE=)

```
- The output of 'go version' : `go version go1.17.1 darwin/amd64`
- Exactly which command you ran: `staticcheck -checks="all","-ST1000","-ST1003","-SA1019" ./...`
- Output of the command and what's wrong with the output: no output, so everything looks fine for staticcheck.

Why there is no version for staticcheck on gha and why he found some issues in the code when local version is fine? On gha it install some dev version or something?

**Comment by dominikh** [comment]
> `GO111MODULE=off go get honnef.co/go/tools/cmd/staticcheck`

This uses Go's GOPATH-mode, which downloads the master branch.

Your local install was installed with the more modern Go modules, which downloaded the latest released version, v2021.1.2.

**Comment by tsknadaj** [comment]
Thanks for answer! Now it's clear for me! :)
